docs/structured-streaming-kafka-integration.md

```diff
@@ -878,7 +878,14 @@ group id, however, please read warnings for this option and use it with caution.
  where to start instead. Structured Streaming manages which offsets are consumed internally, rather
  than rely on the kafka Consumer to do it. This will ensure that no data is missed when new
  topics/partitions are dynamically subscribed. Note that `startingOffsets` only applies when a new
- streaming query is started, and that resuming will always pick up from where the query left off.
+ streaming query is started, and that resuming will always pick up from where the query left off. Note
+ that when the offsets consumed by a streaming application is not in Kafka (e.g., topics are deleted,
+ offsets are out of range, or offsets are removed after offset retention period), the offsets
```

Comment:
"offset retention period" : not sure if the offset is redundant.

Also, perhaps "the offsets are not reset" -> "they will not be reset".